src/meshlink.c

```diff
@@ -216,32 +215,35 @@ struct socket_in_netns_params {
 	int fd;
 };
 
-static void *socket_in_netns_thread(void *arg) {
+/*static void *socket_in_netns_thread(void *arg) {
```

Comment:
You suggested to use conditional compilation `#if LINUX` but I suppose even android satisfies this condition but don't have setns I think so, that's why I commented out the code.

Reply:
I guess the proper thing to do is to add an autoconf test for `setns()` (I think it can just go inside the call to `AC_CHECK_FUNCS` in `configure.ac`), and put the code inside `#ifdef HAVE_SETNS`.

Reply:
Oh, or you can just use `#ifdef CLONE_NEWNET` without having to add an autconf test.

Reply:
Yeah, that's a good idea indeed.